app/src/controllers/client/user/UserController.ts

```diff
@@ -59,7 +64,30 @@ const UserController: UserControllerInterface = {
     if (!user) {
       throw new UnauthorizedError('User not found in request')
     }
-    return reconstructUser(user.id)
+    const u = await UserService.fetchUserWithReservationCountAndReviewCount(user.id)
+    const reservations = await UserService.fetchUserReservationsWithShopMenuAndStylistNames(user.id)
+    return {
+      user: {
+        id: u.id,
+        username: u.username!,
+        email: u.email,
+        lastNameKana: u.lastNameKana,
+        firstNameKana: u.firstNameKana,
+        birthday: u.birthday ? convertDateObjectToOutboundDateString(u.birthday) : undefined,
+        gender: u.gender ? convertEntityGenderToDTOGender(u.gender) : undefined,
+        reservationCount: u.reservationCount,
+        reviewCount: u.reviewCount,
+      },
+      reservations: reservations.map(r => ({
+        id: r.id,
+        shopId: r.shopId,
+        shopName: r.shopName,
+        menuName: r.menuName,
+        status: r.status,
+        reservationDate: convertDateTimeObjectToDateTimeString(r.reservationDate),
+        stylistName: r.StylistName,
+      })),
+    }
```

Comment:
could you plz look at this part once the user part i was trying to return with the reconstruct method but dont know why it gives an error saying cant find username and id basically all the user fields
Had to talk with my dad about school fees so took some time sorry abt that

Reply:
```suggestion
    const reconstructedUser = await reconstructUser(user.id)
    const reservations = await UserService.fetchUserReservationsWithShopMenuAndStylistNames(user.id)
    return {
      user: reconstructedUser,
        id: u.id,
        username: u.username!,
        email: u.email,
        lastNameKana: u.lastNameKana,
        firstNameKana: u.firstNameKana,
        birthday: u.birthday ? convertDateObjectToOutboundDateString(u.birthday) : undefined,
        gender: u.gender ? convertEntityGenderToDTOGender(u.gender) : undefined,
        reservationCount: u.reservationCount,
        reviewCount: u.reviewCount,
      },
      reservations: reservations.map(r => ({
        id: r.id,
        shopId: r.shopId,
        shopName: r.shopName,
        menuName: r.menuName,
        status: r.status,
        reservationDate: convertDateTimeObjectToDateTimeString(r.reservationDate),
        stylistName: r.StylistName,
      })),
    }
```